march_hardware/src/Joint.cpp

```diff
@@ -50,34 +55,62 @@ void Joint::actuateRad(double targetPositionRad)
   this->iMotionCube.actuateRad(targetPositionRad);
 }
 
-double Joint::getAngleRadAbsolute()
+void Joint::readEncoders(const ros::Duration& elapsed_time)
 {
-  if (!hasIMotionCube())
+  const double new_incremental_position = this->iMotionCube.getAngleRadIncremental();
+  const double new_absolute_position = this->iMotionCube.getAngleRadAbsolute();
+
+  // Get velocity from encoder position
+  const double incremental_velocity = (new_incremental_position - this->incremental_position_) / elapsed_time.toSec();
```

Comment:
May be put the 'elapsed_time.toSec()' in a variable as you use it three time.

Reply:
elapsed time is already kind of a variable and toSec is similar to a getter, so it has my preference to leave it like this. Maybe one of the other software boys kan give a third opinion?